File: src/main/java/me/fzzyhmstrs/amethyst_core/interfaces/HitTracking.java
package me.fzzyhmstrs.amethyst_core.interfaces;

import net.minecraft.entity.LivingEntity;
import net.minecraft.item.ItemStack;

public interface HitTracking {
    default void postWearerHit(ItemStack stack, LivingEntity wearer, LivingEntity target){

    }
}

File: src/main/kotlin/me/fzzyhmstrs/amethyst_core/item_util/interfaces/HitTracking.kt
package me.fzzyhmstrs.amethyst_core.item_util.interfaces

import net.minecraft.entity.LivingEntity
import net.minecraft.item.ItemStack

interface HitTracking {
    fun postWearerHit(stack: ItemStack, wearer: LivingEntity, target: LivingEntity)
}